File: Task_Manager_App_Backend/src/main/java/lk/ijse/task_manager_app_backend/entity/Task.java
package lk.ijse.task_manager_app_backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "task")
public class Task {
    @Id
    private Long id;
    private String title;
    private String description;
    private String status;
    private LocalDateTime createdAt;
}

File: Task_Manager_App_Frontend/src/app/models/Task.ts
export class Task {
  id: number;
  title: string;
  description: string;
  status: string;
  createdAt: string;

  constructor(id: number, title: string, description: string, status: string, createdAt: string) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.status = status;
    this.createdAt = createdAt;
  }
}
